compiler/noirc_frontend/src/hir/def_collector/errors.rs

```diff
@@ -291,6 +293,12 @@ impl<'a> From<&'a DefCollectorErrorKind> for Diagnostic {
                 diag
             }
             DefCollectorErrorKind::UnsupportedNumericGenericType(err) => err.into(),
+            DefCollectorErrorKind::TestOnAssociatedFunction { span } => Diagnostic::simple_error(
+                "The `#[test]` attribute may only be used on a non-associated function".into(),
```

Comment:
What's a "non-associated function"? Closures are a sort of function too, but they don't allow `#[test]` either.

```suggestion
                "The `#[test]` attribute is disallowed on `impl` methods".into(),
```
?

Reply:
Sounds good. I copied the message from Rust 😊